### ISSUE TITLE
[@typescript-eslint/no-unused-vars] varsIgnorePattern, wrong name in documentation ?

### ISSUE THREAD

**Issue by mlecoq** [typescript-eslint/typescript-eslint#911]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
            "error",
            { "ignoreRestSiblings": true, "varsIgnorePattern": "^_|^next$" }
        ]
  }
}
```

```TS
 available_remark_items = available_remark_items.concat(
                    Object.entries(remark_types).map(([_, value]) => value.type)
                );
```

**Expected Result**
No error since *_* matches the pattern set in *varsIgnorePattern* according to [documentation](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md#varsignorepattern)

**Actual Result**
An error is raised by eslint on _ (no unused parameter)

**Additional Info**
When I replace  *varsIgnorePattern* by *argsIgnorePattern*, It works fine 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.0.0` |
| `@typescript-eslint/parser`        | `1.10.2` |
| `TypeScript`                       | `3.5.2` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.16.3` |
| `npm`                              | `6.9.0` |

I will make a PR to update doc if you confirm there is an issue

**Comment by mlecoq** [comment]
@bradzacher thanks for your answer ! I didn’t see the second one, why 2 options for the same purpose ? In which case should I use varsIgnorePattern ?

**Comment by mlecoq** [comment]
Ok exception vs argument
